### PR TITLE
[2.11] New Select with Modal component

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

Implements a Select form group with an embedded modal featuring:
* PF4 styles
* Typeahead
* Inside modal:
  * Searches by name, searches on "Enter" keydown
  * Button is disabled until an item is selected
  * Fully paginated
  * When clicking the X, search results are reset (unlike in the gif)

select

modal

**Which issue(s) this PR fixes** 

It's the base for all 2.11 tickets.

### CODE REVIEW

**Review by cntlsn — COMMENTED** [review]
Thank you @josemigallas , I have a couple of questions:
* is a limit set to only display 5 items per page?
* what's the deal with the clearing action on the input field? clicking on the 'X' icon only clears the input field but doesn't actually bring back all the items of the table until the user submits an empty search? if that's the case, then I find it quite unintuitive. if we remove the 'X' icon and its clearing action, would the user still need to enter an empty search, or manually deleting the term in the input field would do the job?

**Comment by josemigallas** [comment]
> Thank you @josemigallas , I have a couple of questions:
> 
> * is a limit set to only display 5 items per page?

Yes, that's the number in the spec. It is not fixed though, 5 is only the default.

> * what's the deal with the clearing action on the input field? clicking on the 'X' icon only clears the input field but doesn't actually bring back all the items of the table until the user submits an empty search? if that's the case, then I find it quite unintuitive. if we remove the 'X' icon and its clearing action, would the user still need to enter an empty search, or manually deleting the term in the input field would do the job?

This has been fixed and now the X (clear button) also resets the search 🙂 .

File: app/javascript/src/Common/components/SelectWithModal.jsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  ]
+
+  // $FlowFixMe $FlowIssue It should not complain since Record.id has union "number | string"
+  const options = getItems(items).map(toSelectOption)
+
+  const handleOnFilter = (e) => {
+    const term = new RegExp(e.target.value, 'i')
+
+    const filteredRecords: T[] = term !== '' ? items.filter(b => term.test(b.name)) : items
+    let filteredItems = filteredRecords.length === 0
```
Comment (didierofrivia) [inline app/javascript/src/Common/components/SelectWithModal.jsx:89]: what's the reason of using `let` here?

**Comment by cntlsn** [comment]
Great, looks good!

> > * is a limit set to only display 5 items per page?
> 
> Yes, that's the number in the spec. It is not fixed though, 5 is only the default.

Ok! but since we're not displaying an interaction to change the number of items per page, we can consider the default value a "fixed" value from a user standpoint, could you confirm?

File: app/javascript/src/Common/components/SelectWithModal.jsx (hunk 1 above)
Comment (josemigallas) [reply app/javascript/src/Common/components/SelectWithModal.jsx:89]: None, I must have left it from an older version.

**Comment by josemigallas** [comment]
> Ok! but since we're not displaying an interaction to change the number of items per page, we can consider the default value a "fixed" value from a user standpoint, could you confirm?

Yes, it is fixed.

**Review by didierofrivia — APPROVED** [review]
LGTM

**Review by guicassolato — APPROVED** [review]
Nice work!

**Review by hallelujah — APPROVED** [review]

**Pull request merged**